### PR TITLE
[DPP-1013] Benchtool: parse more config keys as durations and support more optional config keys

### PR DESCRIPTION
**Pull request by pbatko-da**

### CODE REVIEW

File: ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/config/WorkflowConfig.scala [hunk 1]
```diff
@@ -126,50 +126,44 @@ object WorkflowConfig {
...
-        protected val partyNamePrefixFiltersO: Option[List[PartyNamePrefixFilter]] = None,
+        filters: List[PartyFilter] = List.empty,
+        partyNamePrefixFilters: List[PartyNamePrefixFilter] = List.empty,
         beginOffset: Option[LedgerOffset] = None,
         endOffset: Option[LedgerOffset] = None,
         objectives: Option[StreamConfig.TransactionObjectives] = None,
         subscriptionDelay: Option[FiniteDuration] = None,
         override val maxItemCount: Option[Long] = None,
-        override val timeoutInSecondsO: Option[Long] = None,
+        override val timeoutDurationO: Option[FiniteDuration] = None,
```
Comment (skisel-da) [inline ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/config/WorkflowConfig.scala:136]: Would it be actually better by just calling it `timeout` ? or `cancelStreamTimeout`

File: ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/config/WorkflowConfigParser.scala [hunk 2]
```diff
@@ -31,6 +35,19 @@ object WorkflowConfigParser {
   case class ParserError(details: String)
 
   object Decoders {
+    implicit val scalaDurationDecoder: Decoder[FiniteDuration] =
```
Comment (skisel-da) [inline ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/config/WorkflowConfigParser.scala:38]: there must be a default circe decoder to support FiniteDurations

**Review by skisel-da — APPROVED** [review]

File: ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/config/WorkflowConfigParser.scala (hunk 2 above)
Comment (pbatko-da) [reply ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/config/WorkflowConfigParser.scala:38]: If there is I can't find it.

What I did find though is a duration decoder but for typesafe config, not yaml: https://circe.github.io/circe-config/io/circe/config/syntax$.html

File: ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/config/WorkflowConfig.scala (hunk 1 above)
Comment (pbatko-da) [reply ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/config/WorkflowConfig.scala]: Renamed to `timeoutO`

File: ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/config/WorkflowConfigParser.scala (hunk 2 above)
Comment (skisel-da) [reply ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/config/WorkflowConfigParser.scala:38]: yeah, i could not find it either, so just ignore my comment then :)

**Pull request merged**